File: matlab/task1_5.m
% 1.5. Controle de brilho multiplicativo 
% (valor do pixel resultante = valor do pixel original * c, c real não negativo)
function task1_5()
    assetsDir = ['..' filesep 'assets' filesep];
    filePath = strcat(assetsDir, 'lenna.png');
    
    BRIGHTNESS_ADJUSTMENT = 1.8;
    
    originalImage = imread(filePath);
    modifiedImage = MultiplicativeBrightnessControl(originalImage, BRIGHTNESS_ADJUSTMENT);
        
    % Ploting
    figure();        
    subplot(1,3,1.3), imshow(originalImage,[]), title('Original image')
    subplot(1,3,2.5), imshow(modifiedImage,[]), title('Modified image')
    
    grayscaleImage  = rgb2gray(originalImage);
    modifiedGrayscaleImage = MultiplicativeBrightnessControl(grayscaleImage, BRIGHTNESS_ADJUSTMENT);
    
    figure();        
    subplot(1,3,1.3), imshow(grayscaleImage,[]), title('Original Grayscale')
    subplot(1,3,2.5), imshow(modifiedGrayscaleImage,[]), title('Modified Grayscale')
    
end

function output = MultiplicativeBrightnessControl(input, brightness)
    % MatLab shortcut
    % The pixel type uint8 cannot go higher than 255.
    output = uint8(brightness * input);
end

File: python/task1_5.py
# coding=UTF-8
# 1.5. Controle de brilho multiplicativo
# (valor do pixel resultante = valor do pixel original * c, c real não negativo)

import utils

BRIGHTNESS_ADJUSTMENT = 1.8


def main():
    image = utils.load_image('lenna.png')

    modified_image = image * BRIGHTNESS_ADJUSTMENT
    utils.fit_matrix_in_interval(modified_image)

    utils.display_single_image('Original', image)
    utils.display_single_image('Modified', modified_image)

    utils.wait_key_and_destroy_windows()


if __name__ == "__main__":
    main()